apps/challenges/aws_utils.py

```diff
@@ -835,6 +809,76 @@ def get_logs_from_cloudwatch(
     return logs
 
 
+def get_all_tasks(service_name):
+    """
+    To fetch all tasks for the given service name
+    """
+    if settings.DEBUG:
+        return {
+            "error": False,
+            "details": "Not available for development environment"
+        }
+    else:
+        client = get_boto3_client("ecs", aws_keys)
+        try:
+            tasks_response = client.list_tasks(
+                cluster=COMMON_SETTINGS_DICT["CLUSTER"],
+                serviceName=service_name
+            )
+        except Exception as e:
+            logger.exception(e)
+            return {
+                "error": True,
+                "details": f"Error fetching list of all tasks for the service: {e}"
+            }
+        return {
+            "error": False,
+            "details": tasks_response
+        }
+
+
+def get_all_tasks_status(task_arns):
+    """
+    To fetch status of all containers for the array of task arns
+    """
+    if settings.DEBUG:
+        return {
+            "error": False,
+            "details": "Not available for development environment"
+        }
+    else:
+        try:
+            client = get_boto3_client("ecs", aws_keys)
+            response = client.describe_tasks(
+                cluster=COMMON_SETTINGS_DICT["CLUSTER"],
+                tasks=task_arns
+            )
+            tasks_status = {
+                "error": False,
+                "details": []
+            }
+            task_details = response["tasks"]
+            for task_detail in task_details:
+                containers = []
+                for container in task_detail["containers"]:
+                    containers.append({
+                        "container_arn": container["containerArn"],
+                        "status": container["lastStatus"]
+                    })
+                task_detail = {
+                    "task_arn_def": task_detail["taskArn"],
+                    "containers": containers
+                }
+                tasks_status['details'].append(task_detail)
+            return tasks_status
+        except Exception as e:
+            logger.exception(e)
+            return {
+                "error": True,
+                "details": f"Error fetching worker status: {e}"
```

Comment:
@vinceli1004 can we pass the service name here? I think having service name instead of arns in the log would help in debugging when status fetch fails for any service

Reply:
I am now passing it in as a variable - can you check if that's what you had envisioned (from get_worker_status() in views.py).